_posts/2022-02-10-Grant-Admin-Rights.md

```diff
@@ -0,0 +1,153 @@
+        }
+        Until ($users)
+        Write-Host "User found" -ForegroundColor Yellow
+        return $users.Id
+    }
+}
+function Get-SystemId 
+{
+    param (
+        [Parameter()]
+        [System.String]
+        $UserId
+    )
+    # Get everyone from a group
+    $systemAssociations = Get-JcSdkUserTraverseSystem -UserId $UserId
+    # Systems bound to User
+    $systemsBoundtoUser = @()
+    foreach ($systemAssociation in $systemAssociations) 
+    {
+        $systems = Get-JcSdkSystem -Id $systemAssociation.Id
+
+        #Table of systems
+        $devices = New-Object PSObject -property @{
+            Hostname  = $systems.Hostname
+            OS        = $systems.OS
+            OSVersion = $systems.Version
+            SystemId  = $systemAssociation.Id
+        }
+        $systemsBoundtoUser += $devices
+    }
+
+    # Output the systems bound to user
+    Write-Host "Systems that are bound to user" ($systemsBoundtoUser | Out-String) -ForegroundColor Blue
+    $SystemId = Read-Host "Please enter the SystemId that needs sudo/admin privileges"
+    if ($systemAssociations.Id.Contains($SystemId)) 
+    {
+        Write-Host "System Id found" -ForegroundColor Yellow
+        return $SystemId
+    } 
+    else
+     {
+        Do 
+        {
+            $systemInput = Read-Host "System Id provided not found. Please enter the system id to grant admin rights" 
+            if ($systemInput -eq "exit") 
+            {
+                break;
+            }
+        }
+        Until ($systemAssociations.Id.Contains($systemInput))
+        Write-Host "System found" -ForegroundColor Yellow
+        return $systemInput
+    }
+}
+#Function to elevate permission in x amount of minutes then revokes admin right to user
+Function GrantElevatedPermissions 
+{
+    [CmdletBinding()]
+    param (
+        # The ID of the user to grant elevated permissions
+        [Parameter()]
+        [System.String]
+        $UserId,
+        # The ID of the system to grant a user elevated permissions
+        [Parameter()]
+        [System.String]
+        $SystemId,
+        # Minutes until the user's elevated permission is reverted
+        [Parameter()]
+        [System.Int32]
+        $Mins
+    )
+    # If no sysId then print other systems linked to the User
+    Write-Host "User with Id $UserId will be granted Admin permission on System with Id $SystemId"
+    Write-Host "They will remain admin until $Mins minutes from now..."
+    Write-Warning "Do not close this window until the script is finished or the user will remain admin!"
+    # Calculate the seconds to sleep for 
+    $SleepCalculated = $Mins * 60
+    # Admin attributes
+    $attributes = [JumpCloud.SDK.V2.Models.IGraphOperationSystemAttributes]@{ 'sudo' = @{'enabled' = $true; 'withoutPassword' = $false } }
+    # Set the user to admin
+    Set-JcSdkSystemAssociation -systemid $SystemId -id $UserId -op 'update' -type 'user' -Attributes $attributes.AdditionalProperties
```

Comment:
This line will likely change when we next update our SDKs, just taking a note here more for my own notes